cf_xarray/accessor.py

```diff
@@ -1328,6 +1331,78 @@ def add_bounds(self, dims: Union[Hashable, Iterable[Hashable]]):
 
         return self._maybe_to_dataarray(obj)
 
+    def bounds_to_corners(
+        self,
+        keys: Optional[Union[str, Iterable[str]]] = None,
+        order: Optional[str] = "counterclockwise",
+    ) -> Dataset:
+        """
+        Convert bounds variable to corners.
+
+        There 2 covered cases:
+         - 1D coordinates, with bounds of shape (N, 2),
+           converted to corners of shape (N+1,)
+         - 2D coordinates, with bounds of shape (N, M, 4).
+           converted to corners of shape (N+1, M+1).
+
+
+        Parameters
+        ----------
+        keys : str or Iterable[str], optional
+            The names of the variables whose bounds are to be converted to corners.
+            If not given, converts all available bounds within self.cf.keys().
+        order : {'counterclockwise', 'ccw', 'clockwise', 'cw', None}
+            Valid for 2D coordinates only (bounds of shape NxMx4), ignored otherwise.
+            Order the bounds are given in, assuming that axis0-axis1-upward
```

Comment:
My problem here is that as I understand and experience it, axes order is not super important nor stable in xarray. I mean, if a dataarray was in (i, j) order and then was multiplied to something in (j, i), then xarray will _silently_ output in (j, i) order. This would change the "order" needed for a subsequent `bounds_to_corner` call. 
This is why I implemented the "None" (switching) case first and I would refrain from interpreting carried attributes that could be wrong.

Reply:
Oh I see! Thanks for the clear explanation. Lets mention that as a warning in the docstring.